anx/provider/loadbalancer/reconciliation/reconciliation.go

```diff
@@ -151,16 +143,19 @@ func New(
 //
 // Some resources need others to already exist, so creating all resources returned by ReconcileCheck
 // and then calling ReconcileCheck again will not always result in "nothing to change".
-func (r *reconciliation) ReconcileCheck() ([]types.Object, []types.Object, error) {
-	if err := r.retrieveState(); err != nil {
+func (r *reconciliation) ReconcileCheck(ctx context.Context) ([]types.Object, []types.Object, error) {
```

Comment:
extract `ReconcileCheck` logic to a helper function to be called by `ReconcileCheck` and `Reconcile`, `ReconcileCheck` only sending the `done` signal to the state retriever - this way we don't need the borrow functionality in the state retriever